### PR TITLE
Feature/mx-1347 update workflow triggers

### PR DESCRIPTION
**Pull request by cutoffthetop**
# Added
- create CHANGELOG.md
- configure dependabot updates for github-actions

# Changes
- update workflow triggers and schedules

### CODE REVIEW

File: mex-{{ cookiecutter.project_name }}/.github/dependabot.yml [hunk 1]
```diff
@@ -1,9 +1,16 @@
 version: 2
 updates:
+  - package-ecosystem: "github-actions"
```
Comment (erichesse) [inline mex-{{ cookiecutter.project_name }}/.github/dependabot.yml:3]: so github-action has an internal package-ecosystem and dependabot now keeps track of that as well? How come mex-template needs that enabled but mex-extractor needs not? Has it something to do with the cruft version updating?

File: mex-{{ cookiecutter.project_name }}/.github/workflows/cookiecutter.yml [hunk 2]
```diff
@@ -2,7 +2,9 @@ name: Cookiecutter
 
 on:
   push:
-    branches: ["main"]
+  pull_request:
+  schedule:
+    - cron: '0 0 * * *'
```
Comment (erichesse) [inline mex-{{ cookiecutter.project_name }}/.github/workflows/cookiecutter.yml:7]: So this is a list of conditions and now the workflow triggers on all pushes, PRs and on schedule?

File: mex-{{ cookiecutter.project_name }}/.github/workflows/documentation.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ name: Documentation
 
 on:
   push:
-    branches: ["main"]
+    tags: ["**"]
```
Comment (erichesse) [inline mex-{{ cookiecutter.project_name }}/.github/workflows/documentation.yml:5]: what is that for?

**Review by erichesse — COMMENTED** [review]
Should be fine, but I have a few questions.

File: mex-{{ cookiecutter.project_name }}/.github/workflows/cookiecutter.yml (hunk 2 above)
Comment (cutoffthetop) [reply mex-{{ cookiecutter.project_name }}/.github/workflows/cookiecutter.yml:7]: yes exactly

File: mex-{{ cookiecutter.project_name }}/.github/dependabot.yml (hunk 1 above)
Comment (cutoffthetop) [reply mex-{{ cookiecutter.project_name }}/.github/dependabot.yml:3]: yes the github-actions ecosystem would notify us for example when there is a new version of [actions/checkout@v3](https://github.com/robert-koch-institut/mex-template/blob/feature/mx-1347-update-workflow-triggers/mex-%7B%7B%20cookiecutter.project_name%20%7D%7D/.github/workflows/testing.yml#L19)

> How come mex-template needs that enabled but mex-extractor needs not
as soon as this is merged, i will update the template in mex-extractors using `cruft update` and then these changes will be applied to that repo as well

File: mex-{{ cookiecutter.project_name }}/.github/workflows/documentation.yml (hunk 3 above)
Comment (cutoffthetop) [reply mex-{{ cookiecutter.project_name }}/.github/workflows/documentation.yml:5]: i switched the documentation from building on every push to main to every new tag. that way the documentation is always in sync with the latest release

**Review by erichesse — APPROVED** [review]
Great! Ready to ship

**Pull request merged**